cpp/src/strings/combine.cu

```diff
@@ -192,9 +191,10 @@ std::unique_ptr<column> join_strings(strings_column_view const& strings,
       return bytes;
     },
     thrust::plus<size_type>());
-  CUDA_TRY(cudaMemsetAsync(d_output_offsets, 0, sizeof(size_type), stream.value()));
+  size_type const zero = 0;
+  output_offsets.set_element_async(0, zero, stream);
```

Comment:
Do we need a stream.synchronize() near the end of this function because of the stack variable `zero`, or is something going to ultimately end up causing one before we exit?

Reply:
Excellent question. Turns out the next line below `output_offsets.back_element(stream)` does a synchronize and so we should be good. I discussed this with Mark and he said this was acceptable.

Reply:
Might be worth a comment indicating that if anyone gets in here refactoring things, they should be aware.